mongodb-atlas/cmd/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/icholy/digest"
+	//mongodbatlasplugin "github.com/opencost/opencost-plugins/mongodb-atlas/mongodbatlasplugin"
+)
+
+// TODO move these types to mongodbatlasplugin package
+type CreateCostExplorerQueryPayload struct {
+	Clusters              []string `json:"clusters"`
+	EndDate               string   `json:"endDate"`
+	GroupBy               string   `json:"groupBy"`
+	IncludePartialMatches bool     `json:"includePartialMatches"`
+	Organizations         []string `json:"organizations"`
+	Projects              []string `json:"projects"`
+	Services              []string `json:"services"`
+	StartDate             string   `json:"startDate"`
+}
+
+type CreateCostExplorerQueryResponse struct {
+	token string
+}
+
+type MongoDBAtlasConfig struct {
+	PublicKey  string
+	PrivateKey string
+}
+
+func main() {
+	fmt.Println("Initialize plugin")
+	url := "https://cloud.mongodb.com/api/atlas/v2/orgs/5c06ddb279358e17bfcfca49/billing/costExplorer/usage"
+	payload := CreateCostExplorerQueryPayload{
+
+		EndDate:       "2024-07-01",
+		StartDate:     "2023-12-01",
+		Organizations: []string{"5c06ddb279358e17bfcfca49"},
+	}
+	config := MongoDBAtlasConfig{
+		PublicKey:  os.Getenv("mapuk"),
+		PrivateKey: os.Getenv("maprk"),
+	}
+	resp, error := createCostExplorerQueryToken(url, payload, config)
+	fmt.Println(string(resp))
+	fmt.Println(error)
+}
+
+func getTruth() bool {
+	return true
+}
+
+// pass list of orgs , start date, end date
+func createCostExplorerQueryToken(url string, payload CreateCostExplorerQueryPayload,
+	atlasAuthConfig MongoDBAtlasConfig) (string, error) {
+
+	payloadJson, marshallingError := json.Marshal(payload)
+	if marshallingError != nil {
+		return "error", marshallingError
```

Comment:
golang error handling is one of the most interesting and different parts of the language. Even after 5 years of it being my daily driver, I can't decide if I like it better or not than Java. 

Basically, go never throws exceptions. if you do a nil access, it will panic with a segfault like C, but you can't use exception handling to deal with unexpected errors. you will see error handles like this sprinkled throughout all go code. 

We have found the following approach works:
1. when returning an error, your "good" value should be nil if a pointer, or an empty value if not a pointer.
2. log the error in context (import the logging package from opencost, that will handle logger setup for you, I can point you to examples)
3. wrap the error before you return it. As the error gets passed up the stack, you have nested wrapped messages which function similar to the stacktrace in a java exception. 

Here's how I would handle this error in this case:

```
if marshallingError != nil {
   msg := fmt.Sprintf("createCostExplorerQueryToken: error marshaling JSON payload: %v", err)
   log.Error(msg)
   return "", fmt.Errorf(msg)
}
```

Reply:
Will take point on this.